### PR TITLE
Fix(49525): Adds documentation for the Proxy type.

### LINKED ISSUE #49525: No documentation for the Proxy type
# lib Update Request

## Configuration Check

My compilation *target* is `ES2020` and my *lib* is `the default`.

## Missing / Incorrect Definition
`Proxy`

## Sample Code

The `Proxy` types does not have any documentation. This makes it difficult to use without looking up documentation online

```ts
new Proxy()
```

## Documentation Link

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy

### PR DESCRIPTION
**Pull request by danilolekovic**
Fixes #49525

### CODE REVIEW

File: src/lib/es2015.proxy.d.ts [hunk 1]
```diff
@@ -1,21 +1,114 @@
...
+     * @param target The target callable object.
+     * @param thisArg The `this` argument for the call.
+     * @param argArray The list of arguments for rthe call
+     */
     apply?(target: T, thisArg: any, argArray: any[]): any;
+
+    /**
+     * A trap for the `new` operator.
+     * @param target The target object.
+     * @param argArray The list of arguments for the constructor.
```
Comment (sandersn) [inline src/lib/es2015.proxy.d.ts:13]: I would much rather have a longer doc comment and few comments per parameter. Most of the parameter comments are either trivial as-is, or refer to something that's not defined and would be trivial if there were a good explanation. The reference to `constructor` is a good example of the latter.

File: src/lib/es2015.proxy.d.ts [hunk 2]
```diff
@@ -1,21 +1,114 @@
...
 interface ProxyConstructor {
+    /**
+     * Creates a revocable Proxy object.
+     * @param target A target object to wrap with Proxy.
+     * @param handler An object whose properties define the behavior of Proxy when an operation is attempted on it.
+     */
     revocable<T extends object>(target: T, handler: ProxyHandler<T>): { proxy: T; revoke: () => void; };
+
+    /**
+     * Creates a Proxy object.
```
Comment (sandersn) [inline src/lib/es2015.proxy.d.ts:108]: This is a good place to have an overview of Proxy.

**Review by sandersn — CHANGES_REQUESTED** [review]
I would like to see more prose explanations and fewer short param comments.

File: src/lib/es2015.proxy.d.ts (hunk 1 above)
Comment (danilolekovic) [reply src/lib/es2015.proxy.d.ts:13]: Thank you for the feedback! I have made some changes, please let me know if they are adequate.

Should the trivial short param comments be removed altogether?

Comment (sandersn) [reply src/lib/es2015.proxy.d.ts:13]: Yes, I think so. I'll point out the ones that feel trivial to me and you can let me know what you think.

File: src/lib/es2015.proxy.d.ts [hunk 3]
```diff
@@ -1,21 +1,116 @@
 interface ProxyHandler<T extends object> {
+    /**
+     * A trap method for a function call.
+     * @param target The original callable object which is being proxied.
+     * @param thisArg The `this` argument for the function call.
+     * @param argArray The list of arguments for the function call.
```
Comment (sandersn) [inline src/lib/es2015.proxy.d.ts:6]: thisArg and argArray look trivial

File: src/lib/es2015.proxy.d.ts [hunk 4]
```diff
@@ -1,21 +1,116 @@
...
+     * @param target The original object which is being proxied.
+     * @param argArray The list of arguments for the constructor.
+     * @param newTarget The constructor that was originally called.
+     */
     construct?(target: T, argArray: any[], newTarget: Function): object;
+
+    /**
+     * A trap for `Object.defineProperty()`.
+     * @param target The original object which is being proxied.
+     * @param p The name or `Symbol` of the property whose description is to be retrieved.
```
Comment (sandersn) [inline src/lib/es2015.proxy.d.ts:21]: with the name `property` or `propertyName`, plus the type `string | Symbol` I think this line isn't really needed

File: src/lib/es2015.proxy.d.ts [hunk 5]
```diff
@@ -1,21 +1,116 @@
...
+     * @param attributes The descriptor for the property being defined or modified.
```
Comment (sandersn) [inline src/lib/es2015.proxy.d.ts:22]: same: the name `attributes` plus type `PropertyDescriptor` already says everything the comment does.

File: src/lib/es2015.proxy.d.ts [hunk 6]
```diff
@@ -1,21 +1,116 @@
...
+     * A trap for `Object.preventExtensions()`.
+     * @param target The original object which is being proxied.
+     */
     preventExtensions?(target: T): boolean;
+
+    /**
+     * A trap for setting a property value.
+     * @param target The original object which is being proxied.
+     * @param p The name or `Symbol` of the property to set.
+     * @param value The new value of the property to set.
```
Comment (sandersn) [inline src/lib/es2015.proxy.d.ts:85]: you might be able to get rid of this line too with the name `newValue`

File: src/lib/es2015.proxy.d.ts [hunk 7]
```diff
@@ -1,21 +1,116 @@
...
+     * @param receiver The object to which the assignment was originally directed.
+     * @returns `A `Boolean` indicating whether or not the property was set.
+     */
     set?(target: T, p: string | symbol, value: any, receiver: any): boolean;
+
+    /**
+     * A trap for `Object.setPrototypeOf()`.
+     * @param target The original object which is being proxied.
+     * @param v The object's new prototype or `null`.
```
Comment (sandersn) [inline src/lib/es2015.proxy.d.ts:94]: `v` is a weird name. `newPrototype` is long but descriptive.

**Review by sandersn — CHANGES_REQUESTED** [review]
Pointed out which comments I think are trivial

File: src/lib/es2015.proxy.d.ts (hunk 1 above)
Comment (danilolekovic) [reply src/lib/es2015.proxy.d.ts]: Thank you! I have removed the trivial comments (I agree with all of them being trivial 😃) and added some clearer variable names.

**Review by sandersn — APPROVED** [review]

File: src/lib/es2015.proxy.d.ts (hunk 6 above)
Comment (jason-ha) [reply src/lib/es2015.proxy.d.ts]: FWIW when everything else has a comment it looks like oversight and I wonder if there is anything useful to know. A simple description confirming this is what you think it is good. Doc checkers often will check for completeness. And at least when I read docs I have an idea of parameter order; so, when things are missing in the middle I my brain says something is amiss.

File: src/lib/es2015.proxy.d.ts (hunk 7 above)
Comment (jason-ha) [reply src/lib/es2015.proxy.d.ts]: `v` was left in the method signature. Only doc comment updated to use `newPrototype`.

**Pull request merged**